qiskit/circuit/library/arithmetic/piecewise_chebyshev.py

```diff
@@ -0,0 +1,327 @@
+
+        self._polynomials = None
+        self._poly_r = None
+
+        self.num_state_qubits = num_state_qubits
+
+    def _check_configuration(self, raise_on_failure: bool = True) -> bool:
+        valid = True
+
+        if self._f_x is None:
+            valid = False
+            if raise_on_failure:
+                raise AttributeError('The function to be approximated has not been set.')
+
+        if self._degree is None:
+            valid = False
+            if raise_on_failure:
+                raise AttributeError('The degree of the polynomials has not been set.')
+
+        if self._breakpoints is None:
+            valid = False
+            if raise_on_failure:
+                raise AttributeError('The breakpoints have not been set.')
+
+        if self.num_state_qubits is None:
+            valid = False
+            if raise_on_failure:
+                raise AttributeError('The number of qubits has not been set.')
+
+        if self.num_qubits < self.num_state_qubits + 1:
+            valid = False
+            if raise_on_failure:
+                raise CircuitError('Not enough qubits in the circuit, need at least '
+                                   '{}.'.format(self.num_state_qubits + 1))
+
+        return valid
+
+    @property
+    def f_x(self) -> Callable[[int], float]:
+        """The function to be approximated.
+
+                Returns:
+                    The function to be approximated.
+        """
+        return self._f_x
+
+    @f_x.setter
+    def f_x(self, f_x: Optional[Callable[[int], float]]) -> None:
+        """Set the function to be approximated.
+
+        Note that this may change the underlying quantum register, if the number of state qubits
+        changes.
+
+        Args:
+            f_x: The new function to be approximated.
+        """
+        if self._f_x is None or f_x != self._f_x:
+            self._invalidate()
+            self._f_x = f_x
+
+            self._reset_registers(self.num_state_qubits)
+
+    @property
+    def degree(self) -> int:
+        """The degree of the polynomials.
+
+                Returns:
+                    The degree of the polynomials.
+        """
+        return self._degree
+
+    @degree.setter
+    def degree(self, degree: Optional[int]) -> None:
+        """Set the error tolerance.
+
+        Note that this may change the underlying quantum register, if the number of state qubits
+        changes.
+
+        Args:
+            degree: The new degree.
+        """
+        if self._degree is None or degree != self._degree:
+            self._invalidate()
+            self._degree = degree
+
+            self._reset_registers(self.num_state_qubits)
+
+    @property
+    def breakpoints(self) -> List[int]:
+        """The breakpoints for the piecewise approximation.
+
+                Returns:
+                    The breakpoints for the piecewise approximation.
+        """
+        breakpoints = self._breakpoints
```

Comment:
We should add a check if `num_state_qubits` is None

Reply:
changed in the code